src/strings.rs

```diff
@@ -17,11 +17,49 @@ declare_lint! {
     "Warn on `x = x + ..` where x is a `String`"
 }
 
-#[derive(Copy,Clone)]
+declare_lint! {
+	pub STRING_ADD,
+	Allow,
+	"Warn on `x + ..` where x is a `String`"
+}
+
+#[derive(Copy, Clone)]
 pub struct StringAdd;
 
 impl LintPass for StringAdd {
     fn get_lints(&self) -> LintArray {
+        lint_array!(STRING_ADD)
+    }
+
+    fn check_expr(&mut self, cx: &Context, e: &Expr) {
```

Comment:
You can keep both STRING_ADD and STRING_ADD_ASSIGN, but they can both be emitted from the StringAdd LintPass (like in `methods.MethodsPass`).

It saves a bit of code and keeps related lints together, but mostly it's a style question I guess.